src/RacingGame.js

```diff
@@ -0,0 +1,71 @@
+import { Console } from '@woowacourse/mission-utils';
+import Validation from './Validation.js';
+import ErrorHandler from './ErrorHandler.js';
+
+class RacingGame {
+  constructor(carsList) {
+    this.carsList = carsList;
+    this.winnersList = null;
+  }
+
+  SETTINGS = {
+    gameStatusMark: '-',
+  };
+
+  async getRacingCount() {
+    const racingCount =
+      await Console.readLineAsync('시도할 횟수는 몇 회인가요?');
+    const racingCountNumber = Number(racingCount);
+    const validation = new Validation();
+    const errorHandler = new ErrorHandler();
+    if (!validation.isValidRacingCount(racingCountNumber)) {
+      errorHandler.throwInvalidRacingCountError();
+    }
+
+    return racingCountNumber;
+  }
+
+  playRacing() {
+    this.carsList.forEach((car) => car.move());
+  }
+
+  showGameStatus() {
+    const lastStatusIndex = this.carsList.length - 1;
+    this.carsList.forEach((car, index) => {
+      let statusString = `${car.name} : ${this.SETTINGS.gameStatusMark.repeat(
+        car.location,
+      )}`;
+      if (index === lastStatusIndex) {
+        statusString += '\n';
+      }
+
+      Console.print(statusString);
+    });
+  }
+
+  findWinner() {
+    const initialWinnersArray = [this.carsList[0]];
+    const targetCarsArray = this.carsList.slice(1);
+    const winnersArray = targetCarsArray.reduce((fastestCars, targetCar) => {
+      const targetFastestCar = fastestCars.pop();
+      if (targetFastestCar.location > targetCar.location) {
+        return [...fastestCars, targetFastestCar];
+      }
+
+      if (targetFastestCar.location === targetCar.location) {
+        return [...fastestCars, targetFastestCar, targetCar];
+      }
+
+      return [targetCar];
+    }, initialWinnersArray);
```

Comment:
우와.. reduce 로 한번에 처리하시는 코드 정말 효율적인 것 같아요 👍

fastestCars 는 항상 크기가 1이상일테니 배열에서 pop해오기보다 새로운 차만 넣어주는 방식으로 하면
코드의 혼잡도? 가 조금 줄어들지 않을까 합니닷! ㅎㅎ 



```suggestion
    const winnersArray = targetCarsArray.reduce((fastestCars, targetCar) => {
      const targetFastestCar = fastestCars[0]
      if (targetFastestCar.location > targetCar.location) {
        return [...fastestCars ]; // 기존 그대로 반환
      }

      if (targetFastestCar.location === targetCar.location) {
        return [...fastestCars, targetCar];
      }

      return [targetCar];
    }, initialWinnersArray);
```